log.go

```diff
@@ -80,6 +93,68 @@ type eventLogger struct {
 	// TODO add log-level
 }
 
+func (el *eventLogger) Start(ctx context.Context, name string) context.Context {
+	_, ctx = opentrace.StartSpanFromContext(ctx, name)
+	return ctx
+}
+
+func (el *eventLogger) StartFromParentState(ctx context.Context, name string, parent []byte) context.Context {
+	sc := deserializeContext(parent)
+	_, ctx = opentrace.StartSpanFromContext(ctx, name, otExt.RPCServerOption(sc))
+	return ctx
+}
+
+func (el *eventLogger) SerializeContext(ctx context.Context) []byte {
+	gTracer := opentrace.GlobalTracer()
+	b := make([]byte, 0)
+	carrier := bytes.NewBuffer(b)
```

Comment:
Nit: if you don't need to preallocate this buffer, you can actually just declare it as a stack variable:

```go
var carrier bytes.Buffer
...
... &carrier
```